src/_pytest/config/exceptions.py

```diff
@@ -1,7 +1,3 @@
-from typing import final
-
-
-@final
 class UsageError(Exception):
     """Error in pytest usage or invocation."""
```

Comment:
Let's try to keep this - define `PytestUnhandledCoroutineError` in this file, and then do `UsageError = final(UsageError)` below _that_ to keep the runtime effect.

Reply:
with
```python
UsageError = final(UsageError)
```
mypy gives `error: Cannot assign to a type [misc]` and while it's possible to get around that I'm not sure we should. If a class has `@final` that's a sign that `isinstance(x, UsageError)` will never be `True` for any type other than `UsageError`. I don't know if any of the current type checkers have special handling for final classes, but I think it's reasonable to get rid of it.

If we want runtime effect we can do `setattr(UsageError, '__final__', True)` but I'm likewise not sure that's correct.

Reply:
yeah, let's just not mark it as final then - I'm more excited to turn this into an error than fiddle with typing!

Reply:
actually, I think it's a non-issue because I think it's better to use `fail()` instead of raising `_pytest.config.exceptions.UsageError`.